test/PackageVersionsTest/InstallerTest.php

```diff
@@ -93,6 +93,122 @@ public function testGetSubscribedEvents() : void
+                        'name'    => 'ocramius/package-versions',
+                        'version' => '1.0.0',
+                    ],
+                    [
+                        'name'    => 'foo/bar',
+                        'version' => '1.2.3',
+                        'source'  => ['reference' => 'abc123'],
+                    ],
+                    [
+                        'name'    => 'baz/tab',
+                        'version' => '4.5.6',
+                        'source'  => ['reference' => 'def456'],
+                    ],
+                ],
+                'packages-dev' => [
+                    [
+                        'name'    => 'tar/taz',
+                        'version' => '7.8.9',
+                        'source'  => ['reference' => 'ghi789'],
+                    ],
+                ],
+            ]);
+
+        $repositoryManager->expects(self::any())->method('getLocalRepository')->willReturn($repository);
+
+        $this->composer->expects(self::any())->method('getConfig')->willReturn($config);
+        $this->composer->expects(self::any())->method('getLocker')->willReturn($locker);
+        $this->composer->expects(self::any())->method('getRepositoryManager')->willReturn($repositoryManager);
+        $this->composer->expects(self::any())->method('getPackage')->willReturn($this->getRootPackageMock());
+        $this->composer->expects(self::any())->method('getInstallationManager')->willReturn($installManager);
+
+        $config->expects(self::any())->method('get')->with('vendor-dir')->willReturn($vendorDir);
+
+        Installer::dumpVersionsClass(new Event(
+            'post-install-cmd',
+            $this->composer,
+            $this->io
+        ));
+
+        $expectedSource = <<<'PHP'
+<?php
+
+declare(strict_types=1);
+
+namespace PackageVersions;
+
+/**
+ * This class is generated by ocramius/package-versions, specifically by
+ * @see \PackageVersions\Installer
+ *
+ * This file is overwritten at every run of `composer install` or `composer update`.
+ */
+final class Versions
+{
+    public const ROOT_PACKAGE_NAME = 'root/package';
+    public const VERSIONS          = array (
+  'ocramius/package-versions' => '1.0.0@',
+  'foo/bar' => '1.2.3@abc123',
+  'baz/tab' => '4.5.6@def456',
+  'tar/taz' => '7.8.9@ghi789',
+  'some-replaced/package' => '1.3.5@aaabbbcccddd',
+  'root/package' => '1.3.5@aaabbbcccddd',
+);
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * @throws \OutOfBoundsException If a version cannot be located.
+     */
+    public static function getVersion(string $packageName) : string
+    {
+        if (isset(self::VERSIONS[$packageName])) {
+            return self::VERSIONS[$packageName];
+        }
+
+        throw new \OutOfBoundsException(
+            'Required package "' . $packageName . '" is not installed: cannot detect its version'
+        );
+    }
+}
+
+PHP;
+
+
+        self::assertSame($expectedSource, file_get_contents($expectedFileName));
```

Comment:
I think you only need to check that the file starts with `<?php` here, not the actual contents